core/src/main/java/org/bitcoinj/coinjoin/CoinJoin.java

```diff
@@ -155,20 +160,23 @@ public static boolean isCollateralValid(Transaction txCollateral) {
             }
         }
 
-        for (TransactionInput txin : txCollateral.getInputs()) {
-            Transaction tx = txin.getConnectedTransaction();
-            if (tx != null) {
-                nValueIn = nValueIn.add(tx.getOutput(txin.getOutpoint().getIndex()).getValue());
-            } else {
-                log.info("coinjoin: -- Unknown inputs in collateral transaction, txCollateral={}", txCollateral); /* Continued */
-                return false;
+        if (checkInputs) {
+
+            for (TransactionInput txin : txCollateral.getInputs()) {
+                Transaction tx = txin.getConnectedTransaction();
+                if (tx != null) {
```

Comment:
For CoinJoinServer which is used in tests, the connected transaction won't be available.